### PR TITLE
Run npm install in make-npm.sh

### PR DESCRIPTION
**Pull request by mjethani**
We run `npm install` in `make-npm.sh` if it's running locally.

### CODE REVIEW

**Comment by mjethani** [comment]
@gorhill this removes the `npm install` step every time you do `make lint`.

File: tools/make-npm.sh [hunk 1]
```diff
@@ -35,6 +35,9 @@ else
     mv $tarballname ../uBlock0.npm.tgz
 fi
 ln -sf "$TMPDIR/node_modules"
+if [ -z "$GITHUB_ACTIONS" ]; then
```
Comment (mjethani) [inline tools/make-npm.sh:38]: For reference: https://docs.github.com/en/actions/reference/environment-variables

**Comment by gorhill** [comment]
Actually I have what might be an unrelated question. With all the changes re. `npm` platform lately, what will happen if I do `npm publish` from within `/dist/build/uBlock0.npm/`? I am not sure what will get published now.

**Comment by mjethani** [comment]
> Actually I have what might be an unrelated question. With all the changes re. `npm` platform lately, what will happen if I do `npm publish` from within `/dist/build/uBlock0.npm/`? I am not sure what will get published now.

Try `npm publish --dry-run`.

It should not publish what does not go into the archive via `npm pack`. The same `.npmignore` should be in effect. But always use `--dry-run` to verify first.

**Comment by mjethani** [comment]
> Try [`npm publish --dry-run`](https://docs.npmjs.com/cli/v7/commands/npm-publish#dry-run).

On a fresh system right now after cloning the repo and running `make npm`:

```console
$ npm publish --dry-run
npm notice 
npm notice 📦  @gorhill/ubo-core@0.1.8
npm notice === Tarball Contents === 
npm notice 1.1kB   .eslintrc.json                                      
npm notice 35.1kB  LICENSE.txt                                         
npm notice 5.0kB   README.md                                           
npm notice 1.3kB   build.js                                            
npm notice 240.5kB build/publicsuffixlist.json                         
npm notice 8.5kB   index.js                                            
npm notice 9.1kB   js/base64-custom.js                                 
npm notice 35.0kB  js/biditrie.js                                      
npm notice 14.8kB  js/dynamic-net-filtering.js                         
npm notice 12.1kB  js/filtering-context.js                             
npm notice 2.0kB   js/globals.js                                       
npm notice 9.6kB   js/hnswitches.js                                    
npm notice 28.5kB  js/hntrie.js                                        
npm notice 4.7kB   js/static-filtering-io.js                           
npm notice 109.0kB js/static-filtering-parser.js                       
npm notice 142.4kB js/static-net-filtering.js                          
npm notice 3.2kB   js/text-utils.js                                    
npm notice 5.4kB   js/uri-utils.js                                     
npm notice 10.2kB  js/url-net-filtering.js                             
npm notice 990B    js/wasm/biditrie.wasm                               
npm notice 2.7kB   js/wasm/biditrie.wasm.json                          
npm notice 21.0kB  js/wasm/biditrie.wat                                
npm notice 1.0kB   js/wasm/hntrie.wasm                                 
npm notice 2.8kB   js/wasm/hntrie.wasm.json                            
npm notice 19.2kB  js/wasm/hntrie.wat                                  
npm notice 716B    js/wasm/README.md                                   
npm notice 21.0kB  lib/publicsuffixlist/publicsuffixlist.js            
npm notice 400B    lib/publicsuffixlist/wasm/publicsuffixlist.wasm     
npm notice 1.1kB   lib/publicsuffixlist/wasm/publicsuffixlist.wasm.json
npm notice 9.2kB   lib/publicsuffixlist/wasm/publicsuffixlist.wat      
npm notice 929B    lib/publicsuffixlist/wasm/README.md                 
npm notice 14.7kB  lib/punycode.js                                     
npm notice 411B    lib/regexanalyzer/README.md                         
npm notice 68.8kB  lib/regexanalyzer/regex.js                          
npm notice 1.1kB   package.json                                        
npm notice 1.6kB   test.js                                             
npm notice === Tarball Details === 
npm notice name:          @gorhill/ubo-core                       
npm notice version:       0.1.8                                   
npm notice filename:      @gorhill/ubo-core-0.1.8.tgz             
npm notice package size:  198.8 kB                                
npm notice unpacked size: 844.9 kB                                
npm notice shasum:        6b5ea5730ba5b557e9a14ca810c8db4295448c79
npm notice integrity:     sha512-01VOqE7LPjv9M[...]lku1QIv3xSOIQ==
npm notice total files:   36                                      
npm notice 
+ @gorhill/ubo-core@0.1.8
$ 
```

`npm publish` also takes a tarball. Ideally I would download the tarball built on GitHub and use that with `--dry-run`, if all looks good then publish.

**Comment by mjethani** [comment]
@gorhill if you have time, and this is not important at all, I'd love some help on debugging this:

```diff
diff --git a/platform/npm/tests/request-data.js b/platform/npm/tests/request-data.js
index f1f4bea70..0c015e0f3 100644
--- a/platform/npm/tests/request-data.js
+++ b/platform/npm/tests/request-data.js
@@ -29,12 +29,14 @@ import { createRequire } from 'module';
 import { dirname, resolve } from 'path';
 import { fileURLToPath } from 'url';
 
+import { createWorld } from 'esm-world';
+
 const __dirname = dirname(fileURLToPath(import.meta.url));
 
 const require = createRequire(import.meta.url);
 
-const requests = require('scaling-palm-tree/requests.json');
-const results = require('./data/results.json');
+const requests = require('scaling-palm-tree/requests.json').slice(2990, 3000);
+const results = require('./data/results.json').slice(2990, 3000);
 
 async function read(path) {
     return readFile(resolve(__dirname, path), 'utf8');
@@ -63,7 +65,8 @@ describe('Request data', () => {
     let engine = null;
 
     before(async () => {
-        const module = await import('../index.js');
+        const module = await createWorld('./index.js', { globals: global });
+        //const module = await import('../index.js');
         enableWASM = module.enableWASM;
 
         engine = await module.StaticNetFilteringEngine.create();
@@ -96,7 +99,7 @@ describe('Request data', () => {
 
     // False must go first:
     // https://github.com/gorhill/uBlock/pull/3828#issuecomment-899470383
-    for ( let wasm of [ false, true ] ) {
+    for ( let wasm of [ false ] ) {
         context(`${wasm ? 'Wasm on' : 'Wasm off'}`, () => {
             before(async () => {
                 if ( wasm ) {
```

Apply the above patch to `platform/npm/tests/request-data.js` and run. It fails one test. Then try the same by uncommenting the statement `const module = await import('../index.js');` and commenting out the one above it. It passes all tests.

I wish I knew more about the internals, but this is obviously a problem with the module loader esm-world.

As I said above, this is not important at all. I'll figure it out. I need to write some tests for the module loader anyway so maybe the core issue will get caught there.

**Comment by mjethani** [comment]
> I'd love some help on debugging this

I figured it out.

Files are being loaded multiple times as separate modules. This means there are multiple versions of each module in the same world. There should be only one instance of a module in a given world.

```js
import foo from './example.js';
import bar from './example.js';

assert(foo === bar); // throws
```

I'll fix this.

**Pull request merged**